File: sdk/python/examples/quickstart.py
"""Minimal dr-strange quickstart — run against a `drsg serve` on :7700.

    DRSG_TOKEN=… uv run --project sdk/python python sdk/python/examples/quickstart.py
"""

from drsg import Drsg


def main() -> None:
    db = Drsg()  # base http://127.0.0.1:7700; token from $DRSG_TOKEN

    db.node_create(plane="startup", key="alice", labels=["Person"])
    db.node_create(plane="startup", key="bob", labels=["Person"])
    db.edge_create(plane="startup", src="alice", dst="bob", type="KNOWS")
    db.node_update(plane="startup", key="alice", set={"age": 30})

    alice = db.node_get(plane="startup", key="alice")
    print(f"alice.age = {alice['properties']['age']}")

    stats = db.db_stats()
    print(f"{stats['nodes']} nodes, {stats['edges']} edge(s)")


if __name__ == "__main__":
    main()

File: sdk/java/examples/Quickstart.java
// Minimal dr-strange quickstart — run against a `drsg serve` on :7700.
//   (compile with the built classes + Jackson on the classpath; see sdk/java/README.md)
import io.github.wangyingsm.drsg.Drsg;
import java.util.List;
import java.util.Map;

public class Quickstart {
    public static void main(String[] args) throws Exception {
        Drsg db = new Drsg(); // base http://127.0.0.1:7700; token from $DRSG_TOKEN

        db.nodeCreate(Drsg.NodeCreateParams.of("startup").withKey("alice").withLabels(List.of("Person")));
        db.nodeCreate(Drsg.NodeCreateParams.of("startup").withKey("bob").withLabels(List.of("Person")));
        db.edgeCreate(Drsg.EdgeCreateParams.of("startup", "alice", "bob", "KNOWS"));
        db.nodeUpdate(Drsg.NodeUpdateParams.of("startup").withKey("alice").withSet(Map.of("age", 30)));

        Drsg.NodeRecord alice = db.nodeGet(Drsg.NodeGetParams.of("startup").withKey("alice"));
        System.out.println("alice.age = " + alice.properties().get("age"));

        Drsg.DbStats stats = db.dbStats();
        System.out.println(stats.nodes() + " nodes, " + stats.edges() + " edge(s)");
    }
}
